src/storage/userGoalStorage.ts

```diff
@@ -0,0 +1,40 @@
+import { Period } from "../models/projectModels";
+import { UserGoal } from "../models/userModels";
+
+export async function getUserGoal(): Promise<UserGoal | null> {
+  const goalCountResult = await chrome.storage.sync.get("userGoalCount");
```

Comment:
[nit that doesn’t really matter] I think you can kick these all off at the same time by putting the results in [`Promise.all()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all) and then `await`ing that